lib/pascaltriangle.rb

```diff
@@ -0,0 +1,26 @@
+class Triangle
+
+  def initialize
+    @pascals_row = []
+  end
+
+  def pascal(row_number)
+    if row_number ==  0
+      @pascals_row << [1]
+    else
+      @pascals_row << (0..row_number).map { |i| binomial(row_number, i) }
+    end
+  end
+
+  private
+  def factorial(n)
+    (1..n).inject(:*)
+  end
+
+  private
+  def binomial(n,k)
+    return 1 if n-k <= 0
+    return 1 if k <= 0
+    factorial(n) / (factorial(k) * factorial(n - k))
```

Comment:
@zzokki81 mathematical definition is not always the most optimized way of calculating something. In this particular case, you are not thinking about larger numbers. For them, you are going to have large factorials which you already partially calculated for the previous row. Therefore you are going to have needless calculations. This would be great if you are asked to calculate the 1000th row without going through the first 999 rows.

Can you please switch to a different algorithm, the one that calculates a row based on the previous row?

Also, notice how you didn't use the mathematical definition of the factorial because it uses recursion which is not good programming method in this case. Instead, you used a more program-friendly algorithm.